### DISCUSSION TITLE
Recommended way to extend client code generation

### DISCUSSION THREAD

**Discussion by oliverblaha** [micronaut-projects/micronaut-openapi#1227] [Q&A] [answered]
Hello,

Is there any recommended (and, ideally, documented) way to extend OpenAPI client code generation?

Use case: some slightly nicer integration with Kotlin, starting with adding annotations on the API methods to clarify if its results are nullable or not, and maybe some support for Coroutines. The latter is hopefully also possible with generated Java code that uses a Continuation param, therefore both might actually be achievable by enhancing templates - if I'm lucky :)

I appreciate any hint!

Thanks,
Oliver

**Comment by altro3** [comment, marked as answer]
There are no recommendations, just try generating code with different settings, and check that the resulting code matches what you expect. If there are problems, suggestions or you found bugs, then create an issue: https://github.com/micronaut-projects/micronaut-openapi/issues
